File: zero.Backoffice.UI/old/app/renderers/editors/translation.js

import Editor from 'zero/core/editor.ts';

const editor = new Editor('translation', '@translation.fields.');
editor.blueprintAlias = 'translation';

editor.fieldset(set =>
{
  set.field('key').text(300).required();
  set.field('display').state([
    { label: '@translation.display.text', value: 'text' },
    { label: '@translation.display.html', value: 'html' }
  ]);
});

editor.field('value').rte().required();

export default editor;

File: zero/Localization/Models/Translation.cs
namespace zero.Localization;

public class Translation
{
  /// <summary>
  /// Key of the translation
  /// </summary>
  public string Key { get; set; }
  
  /// <summary>
  /// Value of the translation
  /// </summary>
  public string Value { get; set; }

  /// <summary>
  /// Display + input type
  /// </summary>
  public TranslationDisplay Display { get; set; }
}


public enum TranslationDisplay
{
  Text = 0,
  Html = 1
}